### PR TITLE
Add gallery for tutorials

### PR DESCRIPTION
**Pull request by mloning**
#### Reference Issues/PRs
issue: #1152 
PR: #1205 

#### What does this implement/fix? Explain your changes.
* add nbsphinx gallery for tutorial notebooks

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -224,6 +225,27 @@ def adds(pth):
 nbsphinx_allow_errors = False  # False
 nbsphinx_timeout = 600  # seconds, set to -1 to disable timeout
 
+# since we don't execute the notebooks, we need to manually set the thumbnails here
```
Comment (mgeier) [inline docs/source/conf.py:228]: If you don't actually specify any thumbnails, you might as well not use `nbsphinx_thumbnails` at all. If no thumbnail is specified, the default one will be used.

See https://nbsphinx.readthedocs.io/en/0.8.6/gallery/no-thumbnail.html

And if you *do* want to specify a thumbnail for multiple examples, you can use wildcards, see https://github.com/spatialaudio/nbsphinx/blob/7e20521f44db47bfc72f4502d2782f205f0a3830/doc/conf.py#L29.

Comment (mloning) [reply docs/source/conf.py:228]: Thanks @mgeier - I thought that specifying the thumbnails here would solve the issue with not executing the notebooks but turns out that the issue is something else as explained in https://github.com/spatialaudio/nbsphinx/issues/586

**Comment by mloning** [comment]
@mgeier I think I found the issue. Is it possible that `.. nbgallery::` failed when the notebook name didn't match the notebook in the folder? There was a typo one of the names (see `tutorials.rst` in https://github.com/alan-turing-institute/sktime/pull/1158/commits/0ffc993c8f487ba0c5dcd32cfbde18f068ec40dc). 

There doesn't seem to be any error in the build logs (see https://readthedocs.org/projects/sktime/builds/14347335/).

**Comment by mloning** [comment]
@RNKuhns this is good to go into `docs-dev` from my side, the notebook gallery now works, still some work left in terms of tidying up titles and adding nicer thumbnails, but the basic framework is set up. 

Take a look and let me know what you think. Feel free to merge if you're happy with this going in.

**Comment by mgeier** [comment]
Thanks for the update @mloning!

> Is it possible that `.. nbgallery::` failed when the notebook name didn't match the notebook in the folder?

I tried it with the `nbsphinx` docs, and when using a non-existing document name (in my case the name `non-existing`), I get this warning:

    WARNING: toctree contains reference to nonexisting document 'non-existing'

The remaining documents are *not* shown as thumbnails, but like a "normal" table of contents, just as it was described in https://github.com/spatialaudio/nbsphinx/issues/586.

> There doesn't seem to be any error in the build logs (see https://readthedocs.org/projects/sktime/builds/14347335/).

I wouldn't expect an error, but a warning.

There are 554 warnings in your logs, among them:

```
/home/docs/checkouts/readthedocs.org/user_builds/sktime/checkouts/1158/docs/source/tutorials.rst:17: WARNING: toctree contains reference to nonexisting document 'examples/01a_forecasting_sklearn'
/home/docs/checkouts/readthedocs.org/user_builds/sktime/checkouts/1158/docs/source/tutorials.rst:17: WARNING: toctree contains reference to nonexisting document 'examples/interval_based_classification'
/home/docs/checkouts/readthedocs.org/user_builds/sktime/checkouts/1158/docs/source/tutorials.rst:17: WARNING: toctree contains reference to nonexisting document 'examples/catch22'
/home/docs/checkouts/readthedocs.org/user_builds/sktime/checkouts/1158/docs/source/tutorials.rst:17: WARNING: toctree contains reference to nonexisting document 'examples/minirocket'
/home/docs/checkouts/readthedocs.org/user_builds/sktime/checkouts/1158/docs/source/tutorials.rst:17: WARNING: toctree contains reference to nonexisting document 'examples/forecasting_with_hcrystalball'
/home/docs/checkouts/readthedocs.org/user_builds/sktime/checkouts/1158/docs/source/tutorials.rst:17: WARNING: toctree contains reference to nonexisting document 'examples/dictionary_based_classification'
/home/docs/checkouts/readthedocs.org/user_builds/sktime/checkouts/1158/docs/source/tutorials.rst:17: WARNING: toctree contains reference to nonexisting document 'examples/interpolation'
/home/docs/checkouts/readthedocs.org/user_builds/sktime/checkouts/1158/docs/source/tutorials.rst:17: WARNING: toctree contains reference to nonexisting document 'examples/window_splitter'
```

You should really try to fix all those warnings and then run Sphinx with the `-W` flag.

If there are warnings which you cannot fix, you might be able to silence them with [`suppress_warnings`](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-suppress_warnings).

**Comment by mgeier** [comment]
I've just created https://github.com/spatialaudio/nbsphinx/pull/588 which should show the thumbnails even if there is an invalid document name.

I'm not sure if that's actually good in your case, because it makes it harder to spot the mistake.

**Comment by mloning** [comment]
@mgeier Oh yes, you're right, I only searched for `nbgallery` warnings, thanks for pointing that out! We're working on the warnings.

**Comment by mloning** [comment]
@RNKuhns I've merged this now

**Pull request merged**